src/pages/index.tsx

```diff
@@ -76,7 +79,27 @@ const RequestForm: React.FC = () => {
     fetchData();
   }, []);
 
-  if (loading) return <div>Loading...</div>;
+  if (loading) {
+    return (
+      <main style={{ padding: "1rem" }}>
+        <Navbar />
+        <div className="flex flex-row justify-around mx-10">
+          {Array(3)
+            .fill(null)
+            .map((_, idx) => (
+              <CardSkeleton key={idx} />
+            ))}
+        </div>
+        <div className="flex justify-center mt-4">
+          <Button className="border border-black bg-blue-400 hover:bg-blue-600 text-black w-full md:w-96">
+            Add Event
+          </Button>
+        </div>
+        <IndexPageSkeleton/>
+      </main>
+    );
+  }
+
   if (error) return <div>Error: {error}</div>;
```

Comment:
maybe a page for error would make sense, just redirect them to default next error page?

Reply:
Yep,i think that's a good idea..